### PR TITLE
Mertics instrumentation dbapi

### PR DESCRIPTION
**Pull request by siddhikhapare**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [ ] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [x] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by siddhikhapare** [comment]
Fixes #1158

File: instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py [hunk 1]
```diff
@@ -53,9 +54,15 @@
 )
 from opentelemetry.semconv.trace import SpanAttributes
 from opentelemetry.trace import SpanKind, TracerProvider, get_tracer
+from opentelemetry import metrics
+from opentelemetry.sdk.metrics import MeterProvider
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py:58]: Why is SDK imported here in the instrumentation? Instrumentation never uses SDK.

File: instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py [hunk 2]
```diff
@@ -53,9 +54,15 @@
...
 
 _logger = logging.getLogger(__name__)
 
+meter = MeterProvider().get_meter(__name__)
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py:62]: Why is new meter provider instantiated here?

File: instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py [hunk 3]
```diff
@@ -53,9 +54,15 @@
...
+start_time = time.time()
+end_time = time.time()
+elapsed_time = end_time - start_time
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py:66]: What's happening here?

File: instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py [hunk 4]
```diff
@@ -319,6 +349,39 @@ def get_connection_attributes(self, connection):
         if port is not None:
             self.span_attributes[SpanAttributes.NET_PEER_PORT] = port
 
+database_api_integration = DatabaseApiIntegration(
+    database_system="MySQL",
+    connection_attributes={"database": "mydb", "host": "localhost"}
+)
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py:355]: Why are you creating an instance of DBAPI integration here?

File: instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py [hunk 5]
```diff
@@ -319,6 +349,39 @@ def get_connection_attributes(self, connection):
...
+    value_type=int,
+)
+
+attributes =  database_api_integration.wrapped_cursor_execute(wrapped=None, instance=None, args=None, kwargs=None)
+
+usetime_histogram.record(elapsed_time,attributes=attributes)
+create_time_histogram.record(elapsed_time,attributes=attributes)
+
+pending_requests_updowncounter.add(1)
+pending_requests_updowncounter.subtract(1)
```
Comment (srikanthccv) [inline instrumentation/opentelemetry-instrumentation-dbapi/src/opentelemetry/instrumentation/dbapi/__init__.py:383]: Why are these in global scope? I am not following what is happening here?

**Review by srikanthccv — CHANGES_REQUESTED** [review]
This implementation is incorrect. Please take a look at the other instrumentation for reference. Either close this and reimplement by taking some help or make it draft.

**Comment by siddhikhapare** [comment]
@srikanthccv I will make changes as per your suggestions. I am going through document form here https://opentelemetry-python-contrib.readthedocs.io/en/latest/_modules/opentelemetry/instrumentation/flask.html#FlaskInstrumentor.

**Comment by siddhikhapare** [comment]
@srikanthccv Can you tell me that what to pass as metrics attributes? Do I pass span attributes or connection_attributes?

**Comment by siddhikhapare** [comment]
@srikanthccv As I understood that from given code provides is integration of openTelemetry to trace and monitor database. trace_integration is function that integrates database with otel. The wrap_connect function wraps the connect method of the database to trace database connections. The unwrap_connect function unwraps the previously wrapped connect method. The instrument_connection function instruments an already established database connection to trace and monitor it.
class DatabaseApiIntegration used to wrap database connections, set connection attributes, and trace database. get_traced_connection_proxy function returns a wrapped connection object to be used for tracing. The code also uses a third-party library wrapt to wrap functions. so  Do I get attributes for metrics from connection_attributes to record in histogram? Can you help me in what to use. Thank you.

**Pull request closed without merging**